# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsFrame.cpp
# Commit: 8bf69c3fd814
# Full Hash: 8bf69c3fd81419337cb93f96020bb5e88a333d32
# Author: Jan Varga <Jan.Varga@gmail.com>
# Date: 2012-08-07 13:39:24
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Bug 779684 - Move FrameIteratorFlags to nsFrameIterator class. r=roc
# ==============================================================================

diff -r 78aaaf0d8185 -r 8bf69c3fd814 layout/generic/nsFrame.cpp
--- a/layout/generic/nsFrame.cpp	Mon Aug 06 14:51:33 2012 -0600
+++ b/layout/generic/nsFrame.cpp	Mon Aug 06 20:27:04 2012 -0400
@@ -5782,9 +5782,9 @@
       //resultFrame is not a block frame
       result = NS_ERROR_FAILURE;
 
-      PRUint32 flags = FrameIteratorFlags::FLAG_NONE;
+      PRUint32 flags = nsFrameIterator::FLAG_NONE;
       if (aPos->mScrollViewStop) {
-        flags |= FrameIteratorFlags::FLAG_LOCK_SCROLL;
+        flags |= nsFrameIterator::FLAG_LOCK_SCROLL;
       }
       nsFrameIterator frameTraversal(aPresContext, resultFrame,
                                      ePostOrder, flags);
@@ -5873,9 +5873,9 @@
       if (!found){
         resultFrame = storeOldResultFrame;
 
-        PRUint32 flags = FrameIteratorFlags::FLAG_NONE;
+        PRUint32 flags = nsFrameIterator::FLAG_NONE;
         if (aPos->mScrollViewStop) {
-          flags |= FrameIteratorFlags::FLAG_LOCK_SCROLL;
+          flags |= nsFrameIterator::FLAG_LOCK_SCROLL;
         }
         frameTraversal = nsFrameIterator(aPresContext, resultFrame,
                                          eLeaf, flags);
@@ -6634,12 +6634,12 @@
         return NS_ERROR_FAILURE; //we are done. cannot jump lines
     }
 
-    PRUint32 flags = FrameIteratorFlags::FLAG_FOLLOW_OUT_OF_FLOW;
+    PRUint32 flags = nsFrameIterator::FLAG_FOLLOW_OUT_OF_FLOW;
     if (aScrollViewStop) {
-      flags |= FrameIteratorFlags::FLAG_LOCK_SCROLL;
+      flags |= nsFrameIterator::FLAG_LOCK_SCROLL;
     }
     if (aVisual && presContext->BidiEnabled()) {
-      flags |= FrameIteratorFlags::FLAG_VISUAL;
+      flags |= nsFrameIterator::FLAG_VISUAL;
     }
     nsFrameIterator frameTraversal(presContext, traversedFrame,
                                    eLeaf, flags);